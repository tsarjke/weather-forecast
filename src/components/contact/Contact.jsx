import React from 'react';
import './contact.scss';

import { ReactComponent as GithubIcon } from './img/github.svg';
import { ReactComponent as TgIcon } from './img/telegram2.svg';
import { ReactComponent as LinIcon } from './img/linkedin2.svg';
import { ReactComponent as GmailIcon } from './img/gmail2.svg';

const Contact = () => {
  const contactList = [
    [<GithubIcon />, 'https://github.com/tsarjke'],
    [<TgIcon />, 'https://t.me/tsarjke'],
    [<LinIcon />, 'https://www.linkedin.com/in/tsarev-ivan/'],
    [<GmailIcon />, 'mailto:ivantsarb@gmail.com'],
  ].map(([svg, link]) => (
    <li
      className="contact-list__elem"
      key={link}
    >
      <a
        href={link}
        target="_blank"
        rel="noreferrer"
        className="contact-list__link"
      >
        {svg}
      </a>
    </li>
  ));

  return (
    <ul
      className="contact-list"
      data-testid="contact"
    >
      {contactList}
    </ul>
  );
};

export default Contact;
