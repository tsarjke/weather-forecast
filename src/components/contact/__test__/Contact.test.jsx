import { render, screen } from '@testing-library/react';
import Contact from '../Contact';

describe('Contact', () => {
  test('should render this component without any mistakes', () => {
    render(<Contact />);

    const contactElem = screen.getByTestId('contact');
    expect(contactElem).toMatchSnapshot();
  });
});
