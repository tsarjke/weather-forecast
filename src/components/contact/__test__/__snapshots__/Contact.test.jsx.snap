// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should render this component without any mistakes 1`] = `
<ul
  class="contact-list"
  data-testid="contact"
>
  <li
    class="contact-list__elem"
  >
    <a
      class="contact-list__link"
      href="https://github.com/tsarjke"
      rel="noreferrer"
      target="_blank"
    >
      <svg>
        github.svg
      </svg>
    </a>
  </li>
  <li
    class="contact-list__elem"
  >
    <a
      class="contact-list__link"
      href="https://t.me/tsarjke"
      rel="noreferrer"
      target="_blank"
    >
      <svg>
        telegram2.svg
      </svg>
    </a>
  </li>
  <li
    class="contact-list__elem"
  >
    <a
      class="contact-list__link"
      href="https://www.linkedin.com/in/tsarev-ivan/"
      rel="noreferrer"
      target="_blank"
    >
      <svg>
        linkedin2.svg
      </svg>
    </a>
  </li>
  <li
    class="contact-list__elem"
  >
    <a
      class="contact-list__link"
      href="mailto:ivantsarb@gmail.com"
      rel="noreferrer"
      target="_blank"
    >
      <svg>
        gmail2.svg
      </svg>
    </a>
  </li>
</ul>
`;
